### ISSUE TITLE
Request: offer more options for when you drag&drop files into ScrCpy window

### ISSUE THREAD

**Issue by AndroidDeveloperLB** [Genymobile/scrcpy#5378]
- [V ] I have checked that a similar [feature request](https://github.com/Genymobile/scrcpy/issues?q=is%3Aopen+is%3Aissue+label%3A%22feature+request%22) does not already exist.

**Is your feature request related to a problem? Please describe.**
A bit. Just saving more time in handling things.

**Describe the solution you'd like**
1. Offer a way to avoid installing APK files when dropping them into the window. Maybe by using CTRL while dragging.
2. Offer a way to open the file after it was copied. Maybe by using ALT while dragging.
3. In addition to the auto-installation of APK files, also offer APKS/XAPK/APKM files, which are zipped files that contain the split APK files. 

**Describe alternatives you've considered**
The alternative for just copying the APK file is to zip it and then drag, or to use adb, or to use the file-explorer.
For the rest, it's copying and then handling on the device.

**Comment by archon810** [comment]
Hmm, when I drag and drop an APK file into scrcpy, nothing happens. How do you get it to install APKs, for example?

**Comment by AndroidDeveloperLB** [comment]
@archon810 It installs the APK actually.
Maybe you are talking about APKS/XAPK/APKM files? In these cases, it gets copied .

**Comment by YasarKah** [comment]
Proposal: native drag-and-drop install for XAPK / APKS / APKM bundles

  Right now, dropping an .xapk / .apks / .apkm onto the scrcpy window just pushes it to /sdcard/Download/ as a plain file, since only .apk triggers adb install.

  These formats are all ZIP archives containing a base APK + split APKs (and optionally OBB expansion files). Installing one means running adb install-multiple on the
  split APKs and, if present, pushing the OBB(s) to /sdcard/Android/obb/<package>/. It's a natural extension of the existing drag-to-install-APK feature (#133), triggered
  the same way in file_pusher.

  Before writing a PR, I'd like to check scope and your preferences on two design points:

  1. ZIP reading. The client would need to read the archive. As far as I can tell there's no zip reader in the client today; the lightest option I see is embedding a
  single-header, public-domain library (e.g. miniz). Is adding such a dependency acceptable, or would you prefer a different approach?
  2. Manifest parsing. XAPK bundles carry a manifest.json, but parsing it would pull in a JSON dependency. That can be avoided entirely by working format-agnostically:
  install-multiple every *.apk entry, and for OBB derive the target package from the standard OBB filename ([main|patch].<versionCode>.<package>.obb). This also
  transparently handles .apks bundles (SAI-style), which have no manifest. Would you prefer this no-JSON approach, or is reading manifest.json in scope?

  I've prototyped the full flow (extract → install-multiple → OBB push) and confirmed it works end-to-end on real devices, including split-only bundles and OBB games. I'm
  happy to reimplement it natively in C, cross-platform, following the file_pusher worker pattern.

  Is this something you'd consider merging? If so, are there constraints I should respect (dependencies, splits-only vs. also OBB, etc.)?

**Comment by rom1v** [comment]
Installing apk from scrcpy is implemented because it's just a single `adb install` command. I don't want to more code to unzip or parse files to install for other formats.

**Comment by AndroidDeveloperLB** [comment]
@rom1v That's the third suggestion. What about the others, which are for APK and general files?
Meaning: 

1. Offer a way to avoid installing APK files when dropping them into the window. Maybe by using CTRL while dragging.
2. Offer a way to open the file after it was copied. Maybe by using ALT while dragging. It doesn't have to be XAPK/APKM/APKS/APK file. It can be any file.

Right now, if I have an APK file on the PC, and I only want to copy it, I can't do it directly via dragging to ScrCpy window except for using workarounds to prepare it such as changing file extension or zip it.

**Comment by YasarKah** [comment]
Fair enough @rom1v, that makes sense — I didn't really want to stuff zip handling into scrcpy either.

For anyone here who still wants it though, I went ahead and built it as a separate tool: https://github.com/YasarKah/scrcpy-xapk

It leaves scrcpy's own code alone — just a tiny patch that hands the dropped file off to an external helper, which unzips it and runs install-multiple (plus OBB push for games). Dropping it here in case it saves someone the hassle.
